src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs

```diff
@@ -75,5 +79,20 @@ protected virtual IEnumerable<PropertyInfo> GetRegisteredProperties()
             => AutoRegisteringHelper.ExcludeProperties(
                 typeof(TSourceType).GetProperties(BindingFlags.Public | BindingFlags.Instance).Where(x => x.CanRead),
                 _excludedProperties);
+
+        /// <summary>
+        /// Analyzes a property and returns an instance of <see cref="TypeInformation"/>
+        /// containing information necessary to select a graph type. Any <see cref="GraphQLAttribute"/>
+        /// attributes marked on the property are applied.
+        /// <br/><br/>
+        /// Override this method to enforce specific graph types for specific CLR types, or to implement custom
+        /// attributes to change graph type selection behavior.
+        /// </summary>
+        protected virtual TypeInformation GetTypeInformation(PropertyInfo propertyInfo)
+        {
+            var typeInformation = new TypeInformation(propertyInfo, false);
+            typeInformation.ApplyAttributes();
```

Comment:
Why do you need a separate (and public) method? Maybe call it from `TypeInformation` ctor?

Reply:
Designed so code can be called from AutoRegisteringObjectGraphType between the constructor and applying the attributes. This will be obviously necessary with my next pull request (unless I split it to the one after next).